Tests/AEPOptimizeTests/UnitTests/OptimizePublicAPITests.swift

```diff
@@ -588,8 +588,279 @@ class OptimizePublicAPITests: XCTestCase {
+                              type: EventType.optimize,
+                              source: EventSource.requestContent,
+                              data: testEventData)
+        
+        EventHub.shared.getExtensionContainer(MockExtension.self)?.registerListener(
+            type: EventType.edge,
+            source: EventSource.requestContent) { event in
+                let responseEvent = event.createResponseEvent(
+                    name: "AEP Response Complete",
+                    type: EventType.edge,
+                    source: EventSource.contentComplete,
+                    data:[:]
+                )
+                EventHub.shared.dispatch(event: responseEvent)
+            }
+        
+        EventHub.shared.getExtensionContainer(MockExtension.self)?.registerListener(
+            type: testEvent.type,
+            source: testEvent.source) { event in
+                var eventData: [String: Any] = [:]
+                eventData[OptimizeConstants.EventDataKeys.PROPOSITIONS] = propositionDictionary
+                let timeout: TimeInterval = event.data?[OptimizeConstants.EventDataKeys.TIMEOUT] as? TimeInterval ?? OptimizeConstants.DEFAULT_TIMEOUT
+                let edgeEvent = event.createChainedEvent(name: OptimizeConstants.EventNames.EDGE_PERSONALIZATION_REQUEST,
+                                                         type: EventType.edge,
+                                                         source: EventSource.requestContent,
+                                                         data: eventData)
+                MobileCore.dispatch(event: edgeEvent, timeout: timeout) { _ in
+                    let responseEvent = event.createResponseEvent(
+                        name: "AEP Response Complete",
+                        type: EventType.edge,
+                        source: EventSource.contentComplete,
+                        data: eventData)
+                    EventHub.shared.dispatch(event: responseEvent)
+                }
+            }
+        
+        /// Execute
+        Optimize.updatePropositions(for: [decisionScope], withXdm: nil, andData: nil, timeout: 1) { data, error in
+            XCTAssertNil(error)
+            XCTAssertNotNil(data)
+            
+            /// Verify the decisionScopes
+            let decisionScopes = data?.keys
+            XCTAssertTrue(decisionScopes?.contains(decisionScope) == true)
+
+            /// Verify the propositions
+            let propositions = data?.values
+            XCTAssertNotNil(propositions)
+            XCTAssertFalse(propositions?.isEmpty ?? true)
+            
+            let returnedProposition = data?[decisionScope]
+            XCTAssertNotNil(returnedProposition)
+            XCTAssertEqual(returnedProposition?.id, proposition.id)
+            XCTAssertEqual(returnedProposition?.scope, proposition.scope)
+
+            /// Additional check for propositions
+            if let propositions = propositions {
```

Comment:
since we have to verify on propositions anyway, let's do a `guard` earlier in this block so we don't have worry about unwrapping them later.